### DISCUSSION TITLE
Please report issue to the custom integration author for solax_modbus doing blocking calls at custom_components/solax_modbus/__init__.py, line 318:

### DISCUSSION THREAD

**Discussion by mlduncombe** [wills106/homeassistant-solax-modbus#331] [Q&A]
Solax X1 Gen 4 (H4602)
SolaX Firmware Version Inverter Master 31
SolaX Firmware Version Manager 30

Home Assistant 2023.2.4
Supervisor 2023.01.1
Operating System 9.5
Frontend 20230202.0 - latest

I have recently updated to 2023.2.4 and migrated to an SSD rather than SD on the Pi.

While all seems to be running OK I just noticed the following in the HA logs. do I need to be concerned?

Logger: homeassistant.util.async_
Source: util/async_.py:182
First occurred: 10:32:02 (3172 occurrences)
Last logged: 10:51:03

Detected blocking call to sleep inside the event loop. This is causing stability issues. Please report issue to the custom integration author for solax_modbus doing blocking calls at custom_components/solax_modbus/__init__.py, line 318: return self._client.read_holding_registers(address, count, **kwargs)
Detected blocking call to sleep inside the event loop. This is causing stability issues. Please report issue to the custom integration author for solax_modbus doing blocking calls at custom_components/solax_modbus/__init__.py, line 324: return self._client.read_input_registers(address, count, **kwargs)

**Comment by wills106** [comment]
It's listed under the [known issues](https://github.com/wills106/homeassistant-solax-modbus#known-issues)

You can also hide the errors following: https://github.com/wills106/homeassistant-solax-modbus/issues/329#issuecomment-1429600410

The Serial section of code needs to be updated. But I don't connect to my Inver via RS485, so it's difficult to fix when I can't replicate it / test code out.

**Reply by mlduncombe** [reply]
Thanks,

it was more of a question if its something to worry about long term ? 

All appears to be working well despite the errors and I while I have tried the Waveshare RS485 to Eth (B) route I've not had any luck, not seeing any pwr/link/act LED activity and have 12V on the DC inputs and its my second Waveshare unit. The USB dongle just works, apart from those errors.
